v2/tools/asoctl/internal/crd/cleaner_test.go

```diff
@@ -0,0 +1,311 @@
+	asserter.Expect(crd.Status.StoredVersions).ToNot(ContainElement(alphaVersion))
+	asserter.Expect(crd.Status.StoredVersions).To(ContainElement(betaVersion))
+}
+
+func Test_MigrateDeprecatedCRDResources_NoMigration_IfNotStorage(t *testing.T) {
+	t.Parallel()
+	asserter := NewGomegaWithT(t)
+
+	fakeApiExtClient, fakeClient, cleaner := getClientSets()
+
+	alphaVersion := "v1alpha1api20200601"
+	betaVersion := "v1beta20200601"
+
+	// create CRD
+	definition := newCRDWithStoredVersions(alphaVersion, betaVersion)
+	definition.Spec.Versions = []v1.CustomResourceDefinitionVersion{
+		{
+			Name: alphaVersion,
+		},
+		{
+			Name:    betaVersion,
+			Storage: true,
+		},
+	}
+
+	ns := newNamespace("test-ns")
+
+	_, err := fakeApiExtClient.CustomResourceDefinitions().Create(context.TODO(), definition, metav1.CreateOptions{})
+	asserter.Expect(err).To(BeNil())
+
+	// create Namespace
+	err = fakeClient.Create(context.TODO(), ns)
+	asserter.Expect(err).To(BeNil())
+
+	// create ResourceGroup
+	rg := newResourceGroup("test-rg", ns.Name)
+	err = fakeClient.Create(context.TODO(), rg)
+	asserter.Expect(err).To(BeNil())
+
+	err = cleaner.Run(context.TODO())
+	asserter.Expect(err).To(BeNil())
+
+	crd, err := fakeApiExtClient.CustomResourceDefinitions().Get(context.TODO(), definition.Name, metav1.GetOptions{})
+	asserter.Expect(err).To(BeNil())
+
+	var updatedRG resources.ResourceGroup
+	err = fakeClient.Get(context.TODO(), types.NamespacedName{Name: rg.Name, Namespace: rg.Namespace}, &updatedRG)
+	asserter.Expect(err).To(BeNil())
+	asserter.Expect(updatedRG.ResourceVersion).To(BeEquivalentTo(rg.ResourceVersion))
+
+	asserter.Expect(crd.Status.StoredVersions).ToNot(BeNil())
+	asserter.Expect(crd.Status.StoredVersions).ToNot(BeEquivalentTo(definition.Status.StoredVersions))
+	asserter.Expect(crd.Status.StoredVersions).ToNot(ContainElement(alphaVersion))
+	asserter.Expect(crd.Status.StoredVersions).To(ContainElement(betaVersion))
+}
+
+func Test_CleanDeprecatedCRDVersions_DoesNothing_IfAlphaVersionDoesNotExist(t *testing.T) {
+	t.Parallel()
+	asserter := NewGomegaWithT(t)
+
+	fakeApiExtClient, _, cleaner := getClientSets()
+
+	betaVersion := "v1beta20230101storage"
+
+	definition := newCRDWithStoredVersions(betaVersion)
+
+	_, err := fakeApiExtClient.CustomResourceDefinitions().Create(context.TODO(), definition, metav1.CreateOptions{})
+	if err != nil {
+		return
+	}
+
+	err = cleaner.Run(context.TODO())
+	asserter.Expect(err).To(BeNil())
+
+	crd, err := fakeApiExtClient.CustomResourceDefinitions().Get(context.TODO(), definition.Name, metav1.GetOptions{})
+	asserter.Expect(err).To(BeNil())
+
+	asserter.Expect(crd.Status.StoredVersions).ToNot(BeNil())
+	asserter.Expect(crd.Status.StoredVersions).To(BeEquivalentTo(definition.Status.StoredVersions))
+	asserter.Expect(crd.Status.StoredVersions).To(ContainElement(betaVersion))
+}
+
+func Test_CleanDeprecatedCRDVersions_DoesNothing_BetaVersionDoesNotExist(t *testing.T) {
```

Comment:
minor: Match test name above: 
```suggestion
func Test_CleanDeprecatedCRDVersions_DoesNothing_IfBetaVersionDoesNotExist(t *testing.T) {
```